aiida/repository/backend/abstract.py

```diff
@@ -84,15 +85,43 @@ def put_object_from_file(self, filepath: typing.Union[str, pathlib.Path]) -> str
             return self.put_object_from_filelike(handle)
 
     @abc.abstractmethod
+    def has_objects(self, keys: List[str]) -> List[bool]:
+        """Return whether the repository has an object with the given key.
+
+        :param keys:
+            list of fully qualified identifiers for objects within the repository.
+        :return:
+            list of logicals, in the same order as the keys provided, with value True if the respective
+            object exists and False otherwise.
+        """
+
     def has_object(self, key: str) -> bool:
         """Return whether the repository has an object with the given key.
 
         :param key: fully qualified identifier for the object within the repository.
         :return: True if the object exists, False otherwise.
         """
+        return self.has_objects([key])[0]
+
+    @abc.abstractmethod
+    def list_objects(self) -> Iterator[str]:
```

Comment:
```suggestion
    def list_objects(self) -> Iterable[str]:
```

To clarify: Every iterator is also an iterable, but not every iterable is an iterator.
For example, a list is iterable but a list is not an iterator.
(an iterator can be created from an iterable by using the function `iter()`)